### ISSUE TITLE
Add clojure's partition and partition-all

### ISSUE THREAD

**Issue by Janiczek** [elm-community/elm-list-extra#40]
``` elm
input = [1..5]

# Basically consecutive pairs/triplets/whatever the first argument is.
# If can't get the needed number, stop.
# I'm not sure about the names
partition' 2 input == [[1,2],[3,4]]
partition' 3 input == [[1,2,3]]
partition' 5 input == [[1,2,3,4,5]]
partition' 6 input == []

# the second arg is how much (head sndGroup) advances from (head fstGroup).
# partition' x == partition'' x x
partition'' 2 1 input == [[1,2],[2,3],[3,4],[4,5]]
partition'' 3 3 input == [[1,2,3]] # almost was [[1,2,3],[4,5,6]] but no dice

# also useful
partition-all' 5 input = [[1,2,3,4,5]]
partition-all' 6 input = [[1,2,3,4,5]]

partition-all'' 4 2 input = [[1,2,3,4],[3,4,5]]

# basically get what you can even if you can't get the specified number of them.
```

Definitions in Clojure:
- [partition](https://github.com/clojure/clojure/blob/clojure-1.7.0/src/clj/clojure/core.clj#L3051)
- [partition-all](https://github.com/clojure/clojure/blob/clojure-1.7.0/src/clj/clojure/core.clj#L6947)

I'll add a PR when I get more ti-- some sleep :)

How should it be named? `partition` is already taken.
Something along the lines of `sliding-window`?
